SpatialGDK/Source/SpatialGDKTests/Private/RPCContainer/ObjectSpy.cpp

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "ObjectSpy.h"
+
+#include <Core.h>
+
+TArray<uint8> TypeToArray(ESchemaComponentType Type)
+{
+	// Make sure that if Type value changes - there will be an explicit error
+	check(sizeof(ESchemaComponentType) == sizeof(int32));
```

Comment:
Could this be a `static_assert` ?

Reply:
And if so, just added once at the top of the file.

Reply:
Good idea, will apply this suggestion.

Reply:
I think it's best to keep those asserts in relevant function bodies though, so that if it gets invalidated - it is clear what code is supposed to be changed.

Reply:
Moved it to the top of the file and added a comment.

Reply:
Why not just use sizeof ESchemaComponentType throught the conversion functions instead?

Reply:
The array being created is already dynamically sized.